quickwit/quickwit-metastore/src/metastore/mod.rs

```diff
@@ -891,6 +892,26 @@ impl<T> Default for FilterRange<T> {
     }
 }
 
+/// Maps the given source params to whether checkpoints should be stored in the index metadata
+/// (false) or the shard table (true)
+fn use_shard_api(params: &SourceParams) -> bool {
+    match params {
+        SourceParams::File(FileSourceParams::FileUri(_)) => false,
+        SourceParams::File(FileSourceParams::Sqs(_)) => true,
+        SourceParams::File(FileSourceParams::Stdin) => panic!("stdin cannot be checkpointed"),
+        SourceParams::Ingest => true,
+        SourceParams::IngestApi => false,
+        SourceParams::IngestCli => false,
+        SourceParams::Kafka(_) => false,
+        SourceParams::Kinesis(_) => false,
+        SourceParams::PubSub(_) => false,
+        SourceParams::Pulsar(_) => false,
+        SourceParams::Sqs(_) => true,
```

Comment:
I don't understand. You ended up doing two source params? One via the File source and one via an adhoc `SourceParams`?

Reply:
I'm answering also https://github.com/quickwit-oss/quickwit/pull/5148#discussion_r1662168105 and https://github.com/quickwit-oss/quickwit/pull/5148#discussion_r1662163358 here to provide 1 consistent answer:

This approach is the result our discussion with @guilload:
- The file source reads data from files. The queue (SQS or Pubsub) is just a notification system and is configured as part of the `SourceParams::File`
- If requested by users, we might also provide queue sources (SQS and Pubsub) that read the data directly from the messages. I left the boilerplate for `SourceParams::Sqs` because I find it convenient to keep an eye on how that would look like while I still move the abstractions around. But we can definitively remove it in the name of YAGNI.

Both will use the same infrastructure for message tracking (`QueueCoordinator`), but in terms of API they will be completely separate sources.

Reply:
I removed the `SqsQueue` boilerplate for now. I created a branch to save the boilerplate for reference: https://github.com/quickwit-oss/quickwit/tree/with-sqs-src